src/IssueViz/Editor/LocationTagging/IIssueLocationStore.cs

```diff
@@ -0,0 +1,58 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+using SonarLint.VisualStudio.IssueVisualization.Models;
+
+namespace SonarLint.VisualStudio.IssueVisualization.Editor.LocationTagging
+{
+    public interface IIssueLocationStore
+    {
+        /// <summary>
+        /// Notifies listeners that the set of issues has changed
+        /// </summary>
+        event EventHandler<IssuesChangedEventArgs> IssuesChanged;
+
+        /// <summary>
+        /// Returns all primary and secondary locations for the specified file
+        /// </summary>
+        IEnumerable<IAnalysisIssueLocationVisualization> GetLocations(string filePath);
+
+        /// <summary>
+        /// Used by callers to notify the service that the locations in the specified
+        /// files have been updated
+        /// </summary>
+        void LocationsUpdated(IEnumerable<string> affectedFilePaths);
```

Comment:
It should be `UpdateLocations`, `LocationsUpdated` sounds like an event.

Reply:
The doc comment says what `affectedFilePaths` means. 

`INotifyPropertyChanged` is implemented by the type that is doing the notification i.e. the event source. Here, the service is the sink i.e. it's being notified that something has changed. For me `NotifyLocationsChanged` is more confusing.
I'm happy to rename the method to something else.  I'll change it to `LocationsModified` for now.

We could pass the set of changed locations, but it's more work and not necessary.
The tagger doesn't currently track which locations have changed - it just asks the TextSnapshot to update all of the affected spans.
The service doesn't care which individual locations have changed. It's only interested in knowing the files that have changed so it can tell the appropriate factories to change.

Reply:
`LocationsModified` is a bit problematic, it's an event name, not a function name (same for `LocationsUpdated` for that matter).
Also, the function is supposed to notify the error list, no? In which case, it is the one doing the notification. `NotifyX` sounds right to me, unless I'm missing something.

Reply:
The callers are notifying the locations service that they have updated some of its data.
The interface is the contract between the implementing type and it's callers. What the implementing type does in response to the call is an implementation detail.

Reply:
The API is still off. The callers are notifying the service with seemingly arbitrary data: why just file names? why not the locations list? -- it's because you know that's what you need for the error list, and you know that all the locations service is gonna do is notify the error list with this exact information. Everything relies on the implementation details.
Maybe I'll change my mind when I see how it all comes together, but for now it feels too specific.

Reply:
I'll change the method name to `Refresh` for now.